modules/features2d/src/draw.cpp

```diff
@@ -103,9 +103,13 @@ void drawKeypoints( InputArray image, const std::vector<KeyPoint>& keypoints, In
         {
             cvtColor( image, outImage, COLOR_GRAY2BGR );
         }
+        else if( image.type() == CV_8UC4 )
+        {
+            cvtColor( image, outImage, COLOR_BGRA2BGR );
```

Comment:
It is not needed to force 3 channel images. For example, 4 channel "RGBA" is used on Android.

Update the first "copyTo" condition instead:
```
if (image.type() == CV_8UC3 || image.type() == CV_8UCU)
```

Reply:
If `drawKeypoints` can produce BGRA output images we can expect similar functionality from `drawMatches`. However `drawMatches` takes two inputs and we need to add extra logic to determine output depth.

Reply:
We can add BGRA (4 channels) support into drawMatches too: https://github.com/alalek/opencv/commits/pr13481